### DISCUSSION TITLE
Data for disabled fields is not being validated even with a value

### DISCUSSION THREAD

**Discussion by rohitpotato** [react-hook-form/react-hook-form#2868] [Q&A] [answered]
I have a huge form where some of the fields are disabled but the data is being prefilled from an API using the `reset` API. When I submit the form `react-hook-form` is not running that data (disabled fields data) through the validation and thus causing validation errors. 

This has broken my entire application. I am using version 6.4.0. This started happening after I did `npm install` today.

This is very strange considering everything was working correctly a few days ago. Is this a recent change in the library? If so, is there a fix?

Edit: Value for disabled fields is not being **submitted** as well. This was all working correctly before.

**Comment by bluebill1049** [comment, marked as answer]
I think what you want is `readOnly` instead of `disabled`, this is by following the HTML standard.

https://twitter.com/bluebill1049/status/1300231640392716288

**Reply by rohitpotato** [reply]
Is this a recent change in the library? This would help me understand what I did wrong since everything was working correctly before. I am scratching my head here thinking what went wrong

**Reply by bluebill1049** [reply]
https://github.com/react-hook-form/react-hook-form/issues/2655

good point, i will update that in the changelog, sorry miss that part.

**Reply by bluebill1049** [reply]
updated changelog: https://github.com/react-hook-form/react-hook-form/blob/master/CHANGELOG.md
> 6.5.0
